src/gurt/telemetry.c

```diff
@@ -682,6 +707,101 @@ d_tm_build_path(char **path, char *item, va_list args)
 	return rc;
 }
 
+/**
+ * Compute gauge statistics, min, max, mean, and standard deviation
+ * Uses B. P. Welford's algorithm for computing a running variance.
+ * This implementation computes an integer min and max with floating point
+ * mean and standard deviation.
+ *
+ * \param[in]	node		Pointer to a gauge node
+ */
+void
+d_tm_compute_gauge_stats(struct d_tm_node_t *node)
+{
+	uint64_t	value = 0;
+	double		sum_of_squares = 0.0;
+	double		variance = 0.0;
+	double		mean = 0.0;
+
+	if (node->dtn_metric->dtm_stats == NULL)
+		return;
+
+	value = node->dtn_metric->dtm_data.value;
+	node->dtn_metric->dtm_stats->sample_size++;
+
+	mean = node->dtn_metric->dtm_stats->mean +
+	       ((value - node->dtn_metric->dtm_stats->mean) /
+	       node->dtn_metric->dtm_stats->sample_size);
```

Comment:
Maybe maintaining the sum is enough and we can compute the mean at collect time? Maybe you are worried about overflow?

Reply:
The sum of squares needs the old mean and the new mean, so the mean needs to be computed each time a new metric value is generated.